Nightly SNAPSHOT Failure

Issue:

```
Caused by: java.lang.IllegalStateException: 
This module is configured to use @ConfigMapping annotations but we found a @ConfigRoot without a corresponding @ConfigMapping annotation in: io.quarkus.tika.runtime.TikaConfiguration. 
Either add the annotation or add the -AlegacyConfigRoot=true argument to the annotation processor config in the pom.xml
```

--- a/deployment/src/main/java/io/quarkus/tika/deployment/TikaProcessor.java
+++ b/deployment/src/main/java/io/quarkus/tika/deployment/TikaProcessor.java
@@ -108,9 +108,9 @@ public class TikaProcessor {
     void initializeTikaParser(BeanContainerBuildItem beanContainer, TikaRecorder recorder,
             BuildProducer<ServiceProviderBuildItem> serviceProvider, TikaConfiguration configuration)
             throws Exception {
-        Map<String, List<TikaParserParameter>> parsers = getSupportedParserConfig(configuration.tikaConfigPath,
-                configuration.parsers,
-                configuration.parserOptions, configuration.parser);
+        Map<String, List<TikaParserParameter>> parsers = getSupportedParserConfig(configuration.tikaConfigPath(),
+                configuration.parsers(),
+                configuration.parserOptions(), configuration.parser());
         String tikaXmlConfiguration = generateTikaXmlConfiguration(parsers);
         serviceProvider.produce(new ServiceProviderBuildItem(Parser.class.getName(), parsers.keySet()));
         serviceProvider
@@ -134,10 +134,10 @@ public class TikaProcessor {
         Set<String> providerNames = getProviderNames(Parser.class.getName());
         if (tikaConfigPath.isPresent() || requiredParsers.isEmpty()) {
             return providerNames.stream().filter(pred).collect(Collectors.toMap(Function.identity(),
-                    p -> Collections.<TikaParserParameter> emptyList()));
+                    p -> Collections.emptyList()));
         } else {
             List<String> abbreviations = Arrays.stream(requiredParsers.get().split(",")).map(String::trim)
-                    .collect(Collectors.toList());
+                    .toList();
             Map<String, String> fullNamesAndAbbreviations = abbreviations.stream()
                     .collect(Collectors.toMap(p -> getParserNameFromConfig(p, parserAbbreviations), Function.identity()));
             return providerNames.stream().filter(pred).filter(fullNamesAndAbbreviations::containsKey)
@@ -211,7 +211,7 @@ public class TikaProcessor {
     }
 
     private static String capitalize(String paramName) {
-        if (paramName == null || paramName.length() == 0) {
+        if (paramName == null || paramName.isEmpty()) {
             return paramName;
         }
         char[] chars = paramName.toCharArray();
@@ -236,7 +236,7 @@ public class TikaProcessor {
             Class<?> parserClass = loadParserClass(parserName);
             Method[] methods = parserClass.getMethods();
             String setterMethodName = "set" + capitalize(paramName);
-            String paramType = null;
+            String paramType;
             for (Method method : methods) {
                 if (method.getName().equals(setterMethodName) && method.getParameterCount() == 1) {
                     paramType = method.getParameterTypes()[0].getSimpleName().toLowerCase();

--- a/docs/modules/ROOT/pages/includes/attributes.adoc
+++ b/docs/modules/ROOT/pages/includes/attributes.adoc
@@ -1,4 +1,4 @@
-:quarkus-version: 3.2.9.Final
+:quarkus-version: 3.8.4
 :quarkus-tika-version: 2.0.3
 
 :quarkus-org-url: https://github.com/quarkusio

--- a/docs/modules/ROOT/pages/includes/quarkus-tika.adoc
+++ b/docs/modules/ROOT/pages/includes/quarkus-tika.adoc
@@ -10,7 +10,7 @@ h|[[quarkus-tika_configuration]]link:#quarkus-tika_configuration[Configuration p
 h|Type
 h|Default
 
-a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus.tika.tika-config-path]]`link:#quarkus-tika_quarkus.tika.tika-config-path[quarkus.tika.tika-config-path]`
+a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus-tika-tika-config-path]]`link:#quarkus-tika_quarkus-tika-tika-config-path[quarkus.tika.tika-config-path]`
 
 
 [.description]
@@ -27,7 +27,7 @@ endif::add-copy-button-to-env-var[]
 |
 
 
-a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus.tika.parsers]]`link:#quarkus-tika_quarkus.tika.parsers[quarkus.tika.parsers]`
+a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus-tika-parsers]]`link:#quarkus-tika_quarkus-tika-parsers[quarkus.tika.parsers]`
 
 
 [.description]
@@ -57,7 +57,7 @@ endif::add-copy-button-to-env-var[]
 |
 
 
-a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus.tika.append-embedded-content]]`link:#quarkus-tika_quarkus.tika.append-embedded-content[quarkus.tika.append-embedded-content]`
+a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus-tika-append-embedded-content]]`link:#quarkus-tika_quarkus-tika-append-embedded-content[quarkus.tika.append-embedded-content]`
 
 
 [.description]
@@ -74,7 +74,7 @@ endif::add-copy-button-to-env-var[]
 |`true`
 
 
-a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus.tika.parser-options-parser-options]]`link:#quarkus-tika_quarkus.tika.parser-options-parser-options[quarkus.tika.parser-options]`
+a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus-tika-parser-options-parser-options]]`link:#quarkus-tika_quarkus-tika-parser-options-parser-options[quarkus.tika.parser-options]`
 
 
 [.description]
@@ -96,7 +96,7 @@ endif::add-copy-button-to-env-var[]
 |
 
 
-a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus.tika.parser-parser]]`link:#quarkus-tika_quarkus.tika.parser-parser[quarkus.tika.parser]`
+a|icon:lock[title=Fixed at build time] [[quarkus-tika_quarkus-tika-parser-parser]]`link:#quarkus-tika_quarkus-tika-parser-parser[quarkus.tika.parser]`
 
 
 [.description]

--- a/runtime/src/main/java/io/quarkus/tika/runtime/TikaConfiguration.java
+++ b/runtime/src/main/java/io/quarkus/tika/runtime/TikaConfiguration.java
@@ -3,20 +3,21 @@ package io.quarkus.tika.runtime;
 import java.util.Map;
 import java.util.Optional;
 
-import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
 
 /**
  * Tika parser configuration
  */
+@ConfigMapping(prefix = "quarkus.tika")
 @ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
-public class TikaConfiguration {
+public interface TikaConfiguration {
     /**
      * The resource path within the application artifact to the {@code tika-config.xml} file.
      */
-    @ConfigItem
-    public Optional<String> tikaConfigPath;
+    Optional<String> tikaConfigPath();
 
     /**
      * Comma separated list of the parsers which must be supported.
@@ -40,8 +41,7 @@ public class TikaConfiguration {
      *
      * This property will have no effect if the `tikaConfigPath' property has been set.
      */
-    @ConfigItem
-    public Optional<String> parsers;
+    Optional<String> parsers();
 
     /**
      * Configuration of the individual parsers.
@@ -51,8 +51,7 @@ public class TikaConfiguration {
      * quarkus.tika.parsers = pdf,odf
      * quarkus.tika.parser-options.pdf.sort-by-position = true
      */
-    @ConfigItem
-    public Map<String, Map<String, String>> parserOptions;
+    Map<String, Map<String, String>> parserOptions();
 
     /**
      * Full parser class name for a given parser abbreviation.
@@ -62,8 +61,7 @@ public class TikaConfiguration {
      * quarkus.tika.parsers = classparser
      * quarkus.tika.parser.classparser = org.apache.tika.parser.asm.ClassParser
      */
-    @ConfigItem
-    public Map<String, String> parser;
+    Map<String, String> parser();
 
     /**
      * Controls how the content of the embedded documents is parsed.
@@ -71,6 +69,6 @@ public class TikaConfiguration {
      * Setting this property to false makes the content of each of the embedded documents
      * available separately.
      */
-    @ConfigItem(defaultValue = "true")
-    public boolean appendEmbeddedContent;
+    @WithDefault("true")
+    boolean appendEmbeddedContent();
 }

--- a/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java
+++ b/runtime/src/main/java/io/quarkus/tika/runtime/TikaRecorder.java
@@ -36,20 +36,20 @@ public class TikaRecorder {
         // Create a native Tika Parser. AutoDetectParser is used by default but it is wrapped
         // by RecursiveParserWrapper if the appending of the embedded content is disabled
         Parser nativeParser = new AutoDetectParser(tikaConfig);
-        if (!config.appendEmbeddedContent) {
+        if (!config.appendEmbeddedContent()) {
             // the recursive parser will catch the embedded exceptions by default
             nativeParser = new RecursiveParserWrapper(nativeParser, true);
         }
-        return new TikaParser(nativeParser, config.appendEmbeddedContent);
+        return new TikaParser(nativeParser, config.appendEmbeddedContent());
     }
 
     private static InputStream getTikaConfigStream(TikaConfiguration config, String tikaXmlConfiguration) {
         // Load tika-config.xml resource
         InputStream is;
-        if (config.tikaConfigPath.isPresent()) {
-            is = Thread.currentThread().getContextClassLoader().getResourceAsStream(config.tikaConfigPath.get());
+        if (config.tikaConfigPath().isPresent()) {
+            is = Thread.currentThread().getContextClassLoader().getResourceAsStream(config.tikaConfigPath().get());
             if (is == null) {
-                final String errorMessage = "tika-config.xml can not be found at " + config.tikaConfigPath.get();
+                final String errorMessage = "tika-config.xml can not be found at " + config.tikaConfigPath().get();
                 throw new TikaParseException(errorMessage);
             }
         } else {

